packages/fast-tooling-react/src/css-property-editor/property-editor.tsx

```diff
@@ -0,0 +1,270 @@
+                    onChange={this.handlePropertyValueChange(key)}
+                    value={value}
+                    style={{ width: `${valueWidth}px` }}
+                />
+                ;
+            </div>
+        );
+    }
+
+    private renderNewPropertyInput(
+        keyWidth: number,
+        valueWidth: number
+    ): React.ReactNode {
+        return (
+            <div>
+                <input
+                    type={"text"}
+                    className={this.generateKeyClassNames()}
+                    onChange={this.handleNewPropertyKeyChange}
+                    value={this.state.key}
+                    style={{ width: `${keyWidth}px` }}
+                    ref={this.keyInputRef}
+                />
+                :
+                <input
+                    type={"text"}
+                    className={this.generateValueClassNames()}
+                    onChange={this.handleNewPropertyValueChange}
+                    onKeyDown={this.handleNewPropertyValueKeyDown}
+                    value={this.state.value}
+                    style={{ width: `${valueWidth}px` }}
+                />
+                ;
+            </div>
+        );
+    }
+
+    private generateKeyClassNames(): string {
+        return `${this.props.managedClasses.cssPropertyEditor_input} ${
+            this.props.managedClasses.cssPropertyEditor_inputKey
+        }`;
+    }
+
+    private generateValueClassNames(): string {
+        return `${this.props.managedClasses.cssPropertyEditor_input} ${
+            this.props.managedClasses.cssPropertyEditor_inputValue
+        }`;
+    }
+
+    private getCSSPropertyConfig(data: CSSProperties): CSSPropertiesConfig {
+        const dataConfig: CSSPropertiesConfig = {};
+
+        if (!data) {
+            return dataConfig;
+        }
+
+        Object.keys(data).map(
+            (dataKey: string): void => {
+                dataConfig[dataKey] = {
+                    value: data[dataKey],
+                    keyWidth: 6.6 * dataKey.length,
+                    valueWidth: 6.6 * data[dataKey].length,
+                };
+            }
+        );
+
+        return dataConfig;
+    }
+
+    private handleClick = (e: React.MouseEvent<HTMLPreElement>): void => {
+        if ((e.target as HTMLElement).nodeName !== "INPUT") {
+            this.keyInputRef.current.focus();
+        }
+
+        e.stopPropagation();
+    };
+
+    private handlePropertyKeyChange(
+        oldKey: string
+    ): (e: React.ChangeEvent<HTMLInputElement>) => void {
+        return (e: React.ChangeEvent<HTMLInputElement>): void => {
+            const newData: CSSProperties = {};
+
+            Object.keys(this.props.data).forEach((key: string) => {
+                if (key === oldKey) {
+                    newData[e.target.value] = this.props.data[key];
+                } else {
+                    newData[key] = this.props.data[key];
+                }
+            });
```

Comment:
Sorry if I'm not reading this right, but I don't think you really need a loop here - for the most part you're just copying the object and conditionally changing a single key. I think you can simplify:

```
return this.handleCssUpdate({
  ...this.props.data,
  ...{[e.target.value]: this.props.data[oldKey]}
})
```

Reply:
I am doing this to preserve the location of the keys in the object, this is important because any update to a key would re-order the list and force the current editable key to jump to the bottom. I'm testing for this but I'll add a comment here to clarify.